src/charm.py

```diff
@@ -261,38 +286,116 @@ def _get_gateway_apis_kubernetes_resource_manager(self):
+            workload_tracing_endpoint = self.workload_tracing.get_endpoint("otlp_grpc")
+            if workload_tracing_endpoint:
+                parsed = urlparse(f"//{workload_tracing_endpoint}")
+                providers.append(
+                    {
+                        "name": "otel-tracing",
+                        "opentelemetry": {
+                            "port": parsed.port,
+                            "service": parsed.hostname,
+                        },
+                    }
+                )
+                global_config = {
+                    "meshConfig.enableTracing": "true",
+                    "meshConfig.defaultProviders.tracing": "otel-tracing",
+                    "meshConfig.defaultConfig.tracing.sampling": 100.0,
+                }
+        return providers, global_config
+
+    def _external_authorizers_providers(self) -> List[Dict[str, Any]]:
+        """Return a list of external authorizers provider configurations."""
+        providers = []
+        for relation in self.ingress_config.relations:
+            if self.ingress_config.is_provider_ready(relation):
+                oauth_info = self.ingress_config.get_provider_oauth_info(relation)
+                oauth_name = self.ingress_config.get_oauth_provider_name(relation)
+                if oauth_name and oauth_info:
+                    providers.append(
+                        {
+                            "name": oauth_name,
+                            "envoyExtAuthzHttp": {
+                                "service": oauth_info.oauth_service_name,
+                                "port": oauth_info.oauth_port,
+                                "includeRequestHeadersInCheck": ["authorization", "cookie"],
+                                "headersToUpstreamOnAllow": [
+                                    "authorization",
+                                    "path",
+                                    "x-auth-request-user",
+                                    "x-auth-request-email",
+                                    "x-auth-request-access-token",
+                                ],
+                                "headersToDownstreamOnAllow": ["set-cookie"],
+                                "headersToDownstreamOnDeny": ["content-type", "set-cookie"],
+                            },
+                        }
+                    )
+        return providers
+
+    # TODO: Still a WIP due to unmarshaling issue with istioctl
+    def _helm_list(self, lst: List[str]) -> str:
+        """Convert a list of strings into a Helm-style array literal with quoted items."""
+        return "[" + ",".join(f'"{item}"' for item in lst) + "]"
+
+    # TODO: Still a WIP due to unmarshaling issue with istioctl
+    def _helm_map(self, d: Dict[str, str]) -> str:
+        """Convert a dictionary into a Helm-style map literal with quoted keys and values."""
+        return "{" + ",".join(f'"{k}":"{v}"' for k, v in d.items()) + "}"
```

Comment:
nit: they don't need `self`, so I'd just put them as helper package functions (not methods on the class).  Maybe even move them to a helpers.py